### PR TITLE
Remove C++ linking from oso_prefix feature for now

### PR DESCRIPTION
**Pull request by keith**
Currently in the macOS toolchain C++ linking goes through
`cc_wrapper.sh`, not `wrapped_clang`. I didn't add the handling to
`cc_wrapper` for this argument. These wrappers will likely go away soon,
when they do we should probably make them route to wrapped_clang
instead, in which case we could revert this.

https://github.com/bazelbuild/bazel/pull/12304

### CODE REVIEW

**Comment by aiuto** [comment]
Is this blocked on #12304 or should it happen first?

**Comment by keith** [comment]
Not blocked. We can potentially revert this after that lands, but it depends if that works, and since it sounds like it requires a flag it will take long enough that I think this is worth merging.

**Review by oquenchil — APPROVED** [review]
LGTM

**Pull request closed without merging**
